apps/alert_processor/lib/model/user.ex

```diff
@@ -324,6 +324,9 @@ defmodule AlertProcessor.Model.User do
     |> Repo.transaction()
   end
 
+  @spec get(id()) :: t() | nil
+  def get(id), do: Repo.get_by(__MODULE__, id: id)
```

Comment:
**nitpick (non-blocking):** any reason to use `get_by` rather than just `get` if you're fetching it by the `id`?

Reply:
Restating the questions to confirm that I'm understanding correctly: you are suggesting that in a controller (as a random example), instead of calling `User.get(id)`, I could instead just call `Repo.get_by(User, id: 42)`, is that right?

The honest answer is that I don't think I considered that, so it's an interesting question. Thinking about it now retroactively, there are 3 benefits I can think of for the approach I took:

1. It encapsulates "user stuff" a little more into the domain of the User module
2. Callers don't need to know about `Repo`
3. The syntax is simpler

Are there pros to the other approach I'm not considering?

Alternately, now that I'm looking at at I'm noticing that we currently have the functions `User.get/1` and `User.for_email/1`. Maybe those should be integrated better so maybe you have something like `User.by(id: 42)` and `User.by(email: "jdoe@example.com")`?

Reply:
Oh, hey! Yeah, that's better, I forgot about that one.